xarray/tests/test_strategies.py

```diff
@@ -0,0 +1,278 @@
+        coord_vars = data.draw(coordinate_variables(dim_sizes={"x": 2, "y": 3}))
+
+        arr = data.draw(np_arrays(shape=(2, 3)))
+        da = DataArray(data=arr, coords=coord_vars, dims=["x", "y"])
+        assert isinstance(da, DataArray)
+
+    def test_generates_1d_dim_coords(self):
+        # TODO having a `hypothesis.find(strat, predicate)` function would be very useful here
+        # see https://github.com/HypothesisWorks/hypothesis/issues/3436#issuecomment-1212369645
+        ...
+
+    def test_generates_non_dim_coords(self):
+        ...
+
+
+class TestDataArraysStrategy:
+    @given(dataarrays())
+    def test_given_nothing(self, da):
+        assert isinstance(da, DataArray)
+
+    @given(st.data())
+    def test_given_dims(self, data):
+        da = data.draw(dataarrays(dims=st.just(["x", "y"])))
+        assert da.dims == ("x", "y")
+
+        da = data.draw(dataarrays(dims=st.just({"x": 2, "y": 3})))
+        assert da.sizes == {"x": 2, "y": 3}
+
+    @given(st.data())
+    def test_given_data(self, data):
+        shape = (2, 3)
+        arrs = np_arrays(shape=shape)
+        da = data.draw(dataarrays(data=arrs))
+
+        assert da.shape == shape
+
+    @given(st.data())
+    def test_given_data_and_dims(self, data):
+        arrs = np_arrays(shape=(2, 3))
+        dims = dimension_names(min_dims=2, max_dims=2)
+        da = data.draw(dataarrays(data=arrs, dims=dims))
+        assert da.shape == (2, 3)
+
+        dims = dimension_names(min_dims=3, max_dims=3)
+        with pytest.raises(Unsatisfiable):
+            data.draw(dataarrays(data=arrs, dims=dims))
+
+        arrs = np_arrays(shape=(3, 4))
+        dims = st.just({"x": 3, "y": 4})
+        da = data.draw(dataarrays(data=arrs, dims=dims))
+        assert da.sizes == {"x": 3, "y": 4}
+
+
+class TestDataVariablesStrategy:
+    @given(st.data())
+    def test_given_only_sizes(self, data):
+        dim_sizes = {"x": 2, "y": 3}
+        data_vars = data.draw(data_variables(dim_sizes=dim_sizes))
+        for k, v in data_vars.items():
+            assert isinstance(v, Variable)
+            assert set(v.sizes.items()).issubset(set(dim_sizes.items()))
+
+    @given(st.data())
+    def test_given_restricted_names(self, data):
+        ...
+
+
+class TestDatasetsStrategy:
+    @given(datasets())
+    def test_given_nothing(self, ds):
+        assert isinstance(ds, Dataset)
+
+    @given(st.data())
+    def test_given_data(self, data):
+        dim_sizes = {"x": 3, "y": 4}
+        data_vars = data.draw(data_variables(dim_sizes=dim_sizes))
+        ds = data.draw(datasets(data_vars=st.just(data_vars)))
+        assert set(ds.sizes.items()).issubset(set(dim_sizes.items()))
+
+    @given(st.data())
+    def test_given_dims(self, data):
+        dim_sizes = {"x": 3, "y": 4}
+        ds = data.draw(datasets(dims=st.just(dim_sizes)))
+        assert set(ds.sizes.items()).issubset(set(dim_sizes.items()))
+
+    @settings(deadline=400)
```

Comment:
I had to add this to get the tests to not run out of time locally. I've only had this problem since adding the `attrs` strategy, but I'm not sure how to make that faster.

Reply:
Disabling it is fine, or turning down the maximum size of the generated attributes.

I'll be back in civilisation and up for code review next week.

Reply:
Ah, it looks like the values in the attrs strategy can include some fairly large Numpy arrays - I'd turn max_leaves back up to around ten, and *severely* restrict the size of the generated arrays - e.g. max_size=2, max_dims=3.  That should make it considerably faster, with at most eight elements per array.

Reply:
That seems to work!